pkg/sessionctx/variable/varsutil.go

```diff
@@ -587,3 +589,24 @@ func ParseAnalyzeSkipColumnTypes(val string) map[string]struct{} {
 	}
 	return skipTypes
 }
+
+func parseSchemaCacheSize(s *SessionVars, normalizedValue string, originalValue string) (byteSize uint64, normalizedStr string, err error) {
+	defer func() {
+		if err == nil && byteSize > 0 && byteSize < (512<<20) {
+			s.StmtCtx.AppendWarning(ErrTruncatedWrongValue.FastGenByArgs(TiDBSchemaCacheSize, originalValue))
+			byteSize = 512 << 20
+			normalizedStr = "512MB"
```

Comment:
In fact this is MiB not MB.

And since we already imported the units package, we can use 

```
byteSize = 512 * units.MiB
```

Reply:
fixed